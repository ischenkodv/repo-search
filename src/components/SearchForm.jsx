import React from 'react';
 
class SearchForm extends React.Component {

  constructor() {
    super();

    this.submit = this.submit.bind(this);
  }

  submit(e) {
    e.preventDefault();

    if (!this.props.disabled && typeof this.props.onSubmit === 'function') {
      this.props.onSubmit(this.refs.query.value.trim());
    }
  }

  render() {
    return (
      <form className="form-inline" onSubmit={ this.submit }>
        <div className="form-group">
          <input type="text" name="query" ref="query" className="form-control" placeholder="Enter user name" disabled={ this.props.disabled } />
          <button type="submit" className="btn btn-primary" name="search" disabled={ this.props.disabled }>Search</button>
        </div>
      </form>
    );
  }
}

export default SearchForm;
