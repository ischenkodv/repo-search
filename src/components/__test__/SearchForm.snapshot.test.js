import SearchForm from '../SearchForm';
import renderer from 'react-test-renderer';
import React from 'react';

test('SearchForm component renders the form correctly', () => {
  const fn = function() {};
  const rendered = renderer.create(
    <SearchForm onSubmit={fn} disabled={false} />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});

test('SearchForm can render the form with disabled search input', () => {
  const fn = function() {};
  const rendered = renderer.create(
    <SearchForm onSubmit={fn} disabled={true} />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
