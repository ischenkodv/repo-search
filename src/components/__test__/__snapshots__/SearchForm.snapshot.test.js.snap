exports[`test SearchForm can render the form with disabled search input 1`] = `
<form
  className="form-inline"
  onSubmit={[Function]}>
  <div
    className="form-group">
    <input
      className="form-control"
      disabled={true}
      name="query"
      placeholder="Enter user name"
      type="text" />
    <button
      className="btn btn-primary"
      disabled={true}
      name="search"
      type="submit">
      Search
    </button>
  </div>
</form>
`;

exports[`test SearchForm component renders the form correctly 1`] = `
<form
  className="form-inline"
  onSubmit={[Function]}>
  <div
    className="form-group">
    <input
      className="form-control"
      disabled={false}
      name="query"
      placeholder="Enter user name"
      type="text" />
    <button
      className="btn btn-primary"
      disabled={false}
      name="search"
      type="submit">
      Search
    </button>
  </div>
</form>
`;
